multisig-wallet/src/main.sw

```diff
@@ -0,0 +1,151 @@
+    ///
+    /// let destination_contract = ~Contract::from(/* some b256 here */);
+    /// let value = 100;
+    /// let data = /* some b256 here? */;
+    /// let tx_hash = contract.get_transaction_hash(destination_contract, value, data).call.await.unwrap();
+    ///
+    /// let signatures = [~B512::new(), ~B512::new()];
+    /// contract.executeTransaction(tx_hash, signatures).call().await.unwrap();
+    /// ```
+    fn executeTransaction(tx_hash: b256, signatures: [B512;
+    2]) -> bool {
+        _assert_is_initialized(storage.nonce);
+        _assert_not_executed(get::<bool>(tx_hash));
+
+        // The signers must have increasing values in order to check for duplicates or a zero-value
+        let mut previous_signer: b256 = 0x0000000000000000000000000000000000000000000000000000000000000000;
+
+        // TODO: change to Vec<B512> once implemented and then iterate instead of hardcoding length
+        let mut index = 0;
+        while index < 2 {
+            let signer: Result<Address, EcRecoverError> = ec_recover_address(signatures[index], tx_hash);
+            if let Result::Err = signer {
+                revert(0);
+            };
+
+            let signer = signer.unwrap();
+            _assert_is_owner(get(signer.value));
+            // assert(previous_signer < signer.value);  // TODO: "lt" is not implemented for b256 atm
+            previous_signer = signer.value;
+        }
+
+        // Prevent multiple executions of the same Tx
+        store(tx_hash, true);
+
+        // Increment nonce so that get_transaction_hash() returns a different hash if the same values are used
+        storage.nonce = storage.nonce + 1;
+        log_b256(tx_hash);
+
+        // TODO: Execute (https://github.com/FuelLabs/sway-applications/issues/6 and/or https://github.com/FuelLabs/sway-applications/issues/22)
+
+        true
+    }
+
+    /// Takes in transaction data and hashes it into a unique tx hash
+    ///
+    /// After the hash is generated each signer can sign the hash in order to create
+    /// signatures that can be passed into executeTransaction()
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// // Init contract here ...
+    ///
+    /// let destination_contract = ~Contract::from(/* some b256 here */);
+    /// let value = 100;
+    /// let data = /* some b256 here? */;
+    /// let tx_hash = contract.get_transaction_hash(destination_contract, value, data).call.await.unwrap();
+    /// ```
+    fn get_transaction_hash(to: ContractId, value: u64, data: b256) -> b256 {
+        _assert_is_initialized(storage.nonce);
+        // TODO: data > b256?
+        // TODO: this is probably not following the EIP-191 signing standard. What do?
+        let to_hash = hash_value(to.value, HashMethod::Keccak256);
+        let data_hash = hash_value(data, HashMethod::Keccak256);
+        let value_hash = hash_u64(value, HashMethod::Keccak256);
+        let nonce_hash = hash_u64(storage.nonce, HashMethod::Keccak256);
```

Comment:
EIP-191 _signature_ support will be done in the standard library, separately. But that's orthogonal to how to hash the message that's signed. For that, using SHA-256.